pkg/microservice/aslan/core/common/repository/mongodb/sonar.go

```diff
@@ -0,0 +1,119 @@
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/koderover/zadig/pkg/microservice/aslan/config"
+	"github.com/koderover/zadig/pkg/microservice/aslan/core/common/repository/models"
+	mongotool "github.com/koderover/zadig/pkg/tool/mongo"
+)
+
+type SonarIntegrationColl struct {
+	*mongo.Collection
+
+	coll string
+}
+
+func NewSonarIntegrationColl() *SonarIntegrationColl {
+	name := models.SonarIntegration{}.TableName()
+	return &SonarIntegrationColl{
+		Collection: mongotool.Database(config.MongoDatabase()).Collection(name),
+		coll:       name,
+	}
+}
+
+func (c *SonarIntegrationColl) GetCollectionName() string {
+	return c.coll
+}
+
+func (c *SonarIntegrationColl) EnsureIndex(ctx context.Context) error {
+	return nil
+}
+
+func (c *SonarIntegrationColl) Create(args *models.SonarIntegration) error {
+	if args == nil {
+		return errors.New("sonar integration is nil")
+	}
+
+	_, err := c.InsertOne(context.TODO(), args)
+
+	return err
+}
+
+func (c *SonarIntegrationColl) List(pageNum, pageSize int64) ([]*models.SonarIntegration, int64, error) {
+	query := bson.M{}
+	resp := make([]*models.SonarIntegration, 0)
+	ctx := context.Background()
```

Comment:
If possible, it's better to add `ctx context.Context` argument in the function declaration.